registry/nacos/mcpserver/watcher.go

```diff
@@ -0,0 +1,1013 @@
+		} else {
+			w.watchingService[configKey][serviceUniqueKey] = true
+			mcpServerLog.Infof("subscribe service %v success", serviceUniqueKey)
+		}
+	}
+	return nil
+}
+
+func (w *watcher) getServiceCallback(server *McpServer, configGroup, dataId, path string) func(services []model.Instance, err error) {
+	groupName := server.RemoteServerConfig.ServiceRef.GroupName
+	namespace := server.RemoteServerConfig.ServiceRef.NamespaceId
+	serviceName := server.RemoteServerConfig.ServiceRef.ServiceName
+	protocol := server.Protocol
+
+	if groupName == "DEFAULT_GROUP" {
+		groupName = "DEFAULT-GROUP"
+	}
+	if namespace == "" {
+		namespace = DefaultNacosServiceNamespace
+	}
+	configKey := strings.Join([]string{w.Name, w.NacosNamespace, configGroup, dataId}, DefaultJoiner)
+
+	return func(services []model.Instance, err error) {
+		defer w.UpdateService()
+
+		host := getNacosServiceFullHost(groupName, namespace, serviceName)
+		mcpServerLog.Infof("callback, serviceName : %s", host)
+		if err != nil {
+			mcpServerLog.Errorf("callback error:%v", err)
+			return
+		}
+		serviceEntry := w.generateServiceEntry(host, services)
+		se := &config.Config{
+			Meta: config.Meta{
+				GroupVersionKind: gvk.ServiceEntry,
+				Name:             fmt.Sprintf("%s-%s-%s", IstioMcpAutoGeneratedSeName, configGroup, strings.TrimSuffix(dataId, ".json")),
+				Namespace:        w.namespace,
+			},
+			Spec: serviceEntry,
+		}
+		if protocol == McpSSEProtocol {
+			destinationRule := w.generateDrForSSEService(host)
+			dr := &config.Config{
+				Meta: config.Meta{
+					GroupVersionKind: gvk.DestinationRule,
+					Name:             fmt.Sprintf("%s-%s-%s", IstioMcpAutoGeneratedDrName, configGroup, strings.TrimSuffix(dataId, ".json")),
+					Namespace:        w.namespace,
+				},
+				Spec: destinationRule,
+			}
+			w.cache.UpdateConfigCache(gvk.DestinationRule, configKey, dr, false)
+		}
+		w.cache.UpdateConfigCache(gvk.ServiceEntry, configKey, se, false)
+		vs := w.buildVirtualServiceForMcpServer(serviceEntry, configGroup, dataId, path, server.Name)
+		w.cache.UpdateConfigCache(gvk.VirtualService, configKey, vs, false)
+	}
+}
+
+func (w *watcher) buildVirtualServiceForMcpServer(serviceentry *v1alpha3.ServiceEntry, group, dataId, path, serverName string) *config.Config {
+	if serviceentry == nil {
+		return nil
+	}
+	hosts := w.McpServerExportDomains
+	if len(hosts) == 0 {
+		hosts = []string{"*"}
+	}
+	var gateways []string
+	for _, host := range hosts {
+		cleanHost := common2.CleanHost(host)
+		// namespace/name, name format: (istio cluster id)-host
+		gateways = append(gateways, w.namespace+"/"+
+			common2.CreateConvertedName(w.clusterId, cleanHost),
+			common2.CreateConvertedName(constants.IstioIngressGatewayName, cleanHost))
+	}
+	routeName := fmt.Sprintf("%s-%s-%s", IstioMcpAutoGeneratedHttpRouteName, group, strings.TrimSuffix(dataId, ".json"))
+	baseUrl := w.McpServerBaseUrl
+	if baseUrl == "" {
+		baseUrl = "/mcp-registry"
```

Comment:
There should not be a default value here; mcpServerBaseUrl should be a required option. If the user does not configure it, an error should be thrown directly